### PR TITLE
nios_fixed_address: Remove use_option for name and num

### LINKED ISSUE #67423: Incomplete fix for #64034, numerical options don't work anmyore
##### SUMMARY
#64034 is incomplete, numerical options don't work anmyore

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
nios, lib/ansible/modules/net_tools/nios/nios_fixed_address.py

##### ANSIBLE VERSION
```
ansible 2.9.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/XXXX/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.17 (default, Nov  7 2019, 10:07:09) [GCC 7.4.0]
```

##### CONFIGURATION

```
empty
```

##### OS / ENVIRONMENT
Ubuntu 18.04 with Ansible 2.9.4 from Ansible PPA (2.9.4-1ppa~bionic)

pip freeze|grep -i infoblox
infoblox-client==0.4.23
cherry-picked #64034 applied

##### STEPS TO REPRODUCE

```
    - name: set dhcp options for a ipv4 fixed address
      nios_fixed_address:
        name: hostname.domain
        ipaddr: "{{ vm_ipaddr }}"
        mac: "{{ macaddress }}"
        network: 10.0.0.0/24
        network_view: default
        options:
          - num: "{{ number }}"
            value:  "{{ vm_name }}"
        state: present
        provider: "{{ nios_provider }}"
      connection: local
      register: dhcp_results
```
I suggested a fix in the Infoblox forum, but it needs additional input and checks:
https://community.infoblox.com/t5/API-Integration/Bug-in-Ansible-2-8-with-options-in-nios-fixed-address/m-p/19483#M2522

##### EXPECTED RESULTS

numerical options should work the same way as name options do.

##### ACTUAL RESULTS

```
The full traceback is:
Traceback (most recent call last):
  File "/home/XXXX/.ansible/tmp/ansible-tmp-1581012201.35-244539456118516/AnsiballZ_nios_fixed_address.py", line 102, in <module>
    _ansiballz_main()
  File "/home/XXXX/.ansible/tmp/ansible-tmp-1581012201.35-244539456118516/AnsiballZ_nios_fixed_address.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/XXXX/.ansible/tmp/ansible-tmp-1581012201.35-244539456118516/AnsiballZ_nios_fixed_address.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible.modules.net_tools.nios.nios_fixed_address', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python2.7/runpy.py", line 188, in run_module
    fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 82, in _run_module_code
    mod_name, mod_fname, mod_loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/tmp/ansible_nios_fixed_address_payload_HVTHzC/ansible_nios_fixed_address_payload.zip/ansible/modules/net_tools/nios/nios_fixed_address.py", line 269, in <module>
  File "/tmp/ansible_nios_fixed_address_payload_HVTHzC/ansible_nios_fixed_address_payload.zip/ansible/modules/net_tools/nios/nios_fixed_address.py", line 263, in main
  File "/tmp/ansible_nios_fixed_address_payload_HVTHzC/ansible_nios_fixed_address_payload.zip/ansible/module_utils/net_tools/nios/api.py", line 264, in run
  File "/tmp/ansible_nios_fixed_address_payload_HVTHzC/ansible_nios_fixed_address_payload.zip/ansible/modules/net_tools/nios/nios_fixed_address.py", line 197, in options
KeyError: 'name'
```

### PR DESCRIPTION
**Pull request by Vaishnavi-infoblox**
##### SUMMARY
Fixes: #67423 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/net_tools/nios/nios_fixed_address.py

### CODE REVIEW

File: lib/ansible/modules/net_tools/nios/nios_fixed_address.py [hunk 1]
```diff
@@ -189,12 +189,16 @@ def options(module):
     special_options = ['routers', 'router-templates', 'domain-name-servers',
                        'domain-name', 'broadcast-address', 'broadcast-address-offset',
                        'dhcp-lease-time', 'dhcp6.name-servers']
+    #options-router-templates, broadcast-address-offset, dhcp6.name-servers don't have any associated number
```
Comment (Akasurde) [inline lib/ansible/modules/net_tools/nios/nios_fixed_address.py]:
```suggestion
    # options-router-templates, broadcast-address-offset, dhcp6.name-servers don't have any associated number
```

**Comment by Vaishnavi-infoblox** [comment]
Any update regarding the merge?

**Comment by Akasurde** [comment]
> Any update regarding the merge?

@Vaishnavi-infoblox As of 11 March 2020, Collection migration activity is going on in the ansible/ansible repo. So, repo is frozen until the migration is done. Once this activity is complete we can move forward.

**Comment by Akasurde** [comment]
FYI - https://groups.google.com/forum/#!msg/ansible-devel/oKqgCeYTs-M/cHrOgMw8CAAJ

**Comment by Vaishnavi-infoblox** [comment]
@Akasurde - We are waiting for the merge of this code. Any update?

**Comment by Akasurde** [comment]
@Vaishnavi-infoblox `nios` modules are migrated to community.general collection, but not sure if they need to migrate to their own repo. 

cc @justjais @s-giri @gundalow might have an idea.

**Comment by justjais** [comment]
@Vaishnavi-infoblox currently NIOS content are moved under: [community.general](https://github.com/ansible-collection-migration/community.general), plz check with @s-giri as he has the context of movement, and ideally all of the NIOS content should be moved to infoblox GH repo going forward.

**Comment by saileshgiri** [comment]
Hi @justjais, we'll move NIOS code to infobloxopen/infoblox-ansible. Once done, do we need to raise any PR to remove this content from community.general?

**Comment by Akasurde** [comment]
@saileshgiri Yes, You need to remove content from community.general and update `https://github.com/ansible/ansible/blob/devel/lib/ansible/config/routing.yml`, `https://github.com/ansible/ansible/blob/devel/.github/BOTMETA.yml` and `https://github.com/ansible-collections/community.general/blob/master/.github/BOTMETA.yml`

Let me know if you need any help.

**Comment by gundalow** [comment]
You'll also need to move the following from community.general to your new repo 
* `tests/integration/nios`
* any nios lines from `tests/sanity/ignore*.txt`


PR1: ansible/ansible
PR2: Remove content from community.general
PR3: Add content to new collection repo

**Comment by gundalow** [comment]
I've improved the documentation on moving content between repos, please see https://github.com/ansible-collections/overview/blob/master/README.rst#q-what-should-i-do-to-move-plugins-across-collections-during-migration

**Comment by gundalow** [comment]
@Vaishnavi-infoblox @saileshgiri Hi,
Are the next steps for you to take clear?
If not, please let us know.

**Comment by saileshgiri** [comment]
Hi @gundalow @Akasurde, thank you for sharing the details. We're working on this and we'll reach out in case we need any more information or clarification.

**Comment by Vaishnavi-infoblox** [comment]
Hi @gundalow @Akasurde, 
I have raised a new PR(https://github.com/ansible/ansible/pull/69236), with the following changes
•	sanity ignore lines from tests/sanity/ignore*.txt
•	remove entries from .github/BOTMETA.yml
•	Update meta/routing.yml to contain redirects for EVERY PLUGIN, pointing to the new collection name.
•	integration tests: tests/integrations/targets/

We have added those contents in our infoblox-ansible repo. Please review and let me know if I missed any.

**Comment by danielmellado** [comment]
Thank you very much for your interest in Ansible. This plugin is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general
Please re-submit this issue in the above repository.
If you have further questions please stop by IRC or the mailing list:

IRC: #ansible-network on irc.freenode.net
mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**